### DISCUSSION TITLE
About the current state of swigm3

### DISCUSSION THREAD

**Discussion by nsgnkhibdk2cls0f** [modula3/cm3#1145] [General]
@demoitem You should add to README that it's not yet in a usable form. It took me many hours downloading and installing MSYS2 with all of the packages needed to bootstrap swigm3 and CM3 (I normally use a MSYS2-less setup of CM3 thanks to @jpgpng), only to realize it doesn't work yet. It did produce .i3 and .m3, but inside them, it's Java code, not M3 code. It seems you are modifying Java backend to create M3 backend and it's in a very WIP state. For anyone want to use swig with M3, you should use the old m3swig instead.

**Comment by 1997tn** [comment]
If you read my comment you will not wasted your time:

https://github.com/modula3/cm3/discussions/1140#discussioncomment-4699264

**Comment by VictorMiasnikov** [comment]
Small question:

 Did You test "SWIG for Modula-3" version 4.11?

https://github.com/demoitem/swigm3/commit/ee7e848db24938390c76c59cc14f3a49c42a6cc8

( I don't sure about number...)

**Comment by VictorMiasnikov** [comment]
} m3swig

Why You think, what m3swig is "old"?

**Comment by VictorMiasnikov** [comment]
} it's Java code, not M3 code. 

Please, see inside LLVM10 and ( or) LLVM10BINDING folders fully worked example of using m3swig.

P.S. I hope, I have public m3swig binaries inside one of Modula-3 releases.

 You can use this binary file

**Comment by 1997tn** [comment]
@VictorMiasnikov This thread is about swigm3, not m3swig.

**Reply by VictorMiasnikov** [reply]
} swigm3, not m3swig

They are one software,  but different SWIG version based

**Reply by 1997tn** [reply]
swigm3 is a rewrite from scratch. He used Java backend as the reference for his M3 backend. It's still in the very early stage and not working yet.

**Comment by VictorMiasnikov** [comment]
= =
swigm3 is a rewrite from scratch. He used Java backend as the reference for his M3 backend.
= =

I recheck this info ( Ok: try recheck)

= =
 It's still in the very early stage and not working yet.
= =
  If 4.11 (?) don't work now it's not be a big problem.

We have worked version

**Reply by 1997tn** [reply]
> We have worked version

If m3swig is not failed to process some C headers with the bug `Assertion failed: b->refcount > 0, file DOH/base.c, line 42`, he will not have to rebase to latest swig and reimplement it from scratch based on the Java backend.

**Reply by VictorMiasnikov** [reply]
> process some C headers

Please, process LLVM v9 or LLVM v10 headers.

(

" C headers with the bug " are special example for demonstration bug in SWIG?
I.e. they don't usefull for practical purposes?

)
